CMakeLists.txt

```
@@ -157,9 +163,9 @@ separate build directory. Note: Please remove autogenerated file \
 `CMakeCache.txt` and directory `CMakeFiles` in the current directory.")
 endif()
 
-if(SLEEF_ENABLE_LTO AND BUILD_SHARED_LIBS)
-  message(FATAL_ERROR "SLEEF_ENABLE_LTO and BUILD_SHARED_LIBS cannot be specified at the same time")
-endif(SLEEF_ENABLE_LTO AND BUILD_SHARED_LIBS)
+if(SLEEF_ENABLE_LTO AND _INTERNAL_SLEEF_BUILD_SHARED_LIBS)
+  message(FATAL_ERROR "SLEEF_ENABLE_LTO and _INTERNAL_SLEEF_BUILD_SHARED_LIBS cannot be specified at the same time")
+endif(SLEEF_ENABLE_LTO AND _INTERNAL_SLEEF_BUILD_SHARED_LIBS)
```

With my other suggestions:

```suggestion
if (SLEEF_ENABLE_LTO AND BUILD_SHARED_LIBS)
  message(FATAL_ERROR "SLEEF_ENABLE_LTO and (SLEEF_)BUILD_SHARED_LIBS cannot be enabled at the same time")
endif ()
```
Note that duplicating the condition in the `endif` hasn't been a requirement since at _least_ 2.8.x but it might be as far back as 2.4.x. Definitely not needed in 3.18, which is your minimum.